### PR TITLE
deprecate DenyExecOnPrivileged/DenyEscalatingExec

### PR DESCRIPTION
**Pull request by liggitt**
Update guidance for admission plugins

/hold
for https://github.com/kubernetes/kubernetes/pull/72737

### CODE REVIEW

File: content/en/blog/_posts/2016-08-00-Security-Best-Practices-Kubernetes-Deployment.md [hunk 1]
```diff
@@ -205,7 +205,15 @@ Reference [here](http://kubernetes.io/docs/api-reference/v1/definitions/#_v1_pod
```
Comment (liggitt) [inline content/en/blog/_posts/2016-08-00-Security-Best-Practices-Kubernetes-Deployment.md]: wasn't sure how revisionist we get about blog post updates, but this advice isn't current

Comment (bgrant0607) [reply content/en/blog/_posts/2016-08-00-Security-Best-Practices-Kubernetes-Deployment.md]: The ResourceQuota section is incorrect, also

Comment (bgrant0607) [reply content/en/blog/_posts/2016-08-00-Security-Best-Practices-Kubernetes-Deployment.md]: Maybe just mark the blog post as being out of date and containing errors?

File: content/en/blog/_posts/2016-08-00-Security-Best-Practices-Kubernetes-Deployment.md [hunk 2]
```diff
@@ -205,7 +205,15 @@ Reference [here](http://kubernetes.io/docs/api-reference/v1/definitions/#_v1_pod
 
 
 
-In case you are running containers with elevated privileges (--privileged) you should consider using the “DenyEscalatingExec” admission control. This control denies exec and attach commands to pods that run with escalated privileges that allow host access. This includes pods that run as privileged, have access to the host IPC namespace, and have access to the host PID namespace. For more details on admission controls, see the Kubernetes [documentation](http://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/).
+By default, users with the ability to create pods can make use of features that allow root-level 
+access to the nodes they run on, including mounting arbitrary host paths, running as root, and
+running privileged containers.
+
+If you do not want all users with the ability to create a pod to have root-level access to nodes,
+you should consider using a policy-based admission plugin to limit the types of pods that can be created.
```
Comment (zparnold) [inline content/en/blog/_posts/2016-08-00-Security-Best-Practices-Kubernetes-Deployment.md]:
```suggestion
you should consider using a policy-based admission plugin to limit the types of Pods that can be created.
```

File: content/en/blog/_posts/2016-08-00-Security-Best-Practices-Kubernetes-Deployment.md [hunk 3]
```diff
@@ -205,7 +205,15 @@ Reference [here](http://kubernetes.io/docs/api-reference/v1/definitions/#_v1_pod
 
 
 
-In case you are running containers with elevated privileges (--privileged) you should consider using the “DenyEscalatingExec” admission control. This control denies exec and attach commands to pods that run with escalated privileges that allow host access. This includes pods that run as privileged, have access to the host IPC namespace, and have access to the host PID namespace. For more details on admission controls, see the Kubernetes [documentation](http://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/).
+By default, users with the ability to create pods can make use of features that allow root-level 
+access to the nodes they run on, including mounting arbitrary host paths, running as root, and
+running privileged containers.
+
+If you do not want all users with the ability to create a pod to have root-level access to nodes,
```
Comment (zparnold) [inline content/en/blog/_posts/2016-08-00-Security-Best-Practices-Kubernetes-Deployment.md]:
```suggestion
If you do not want all users with the ability to create a Pod to have root-level access to nodes,
```

File: content/en/blog/_posts/2016-08-00-Security-Best-Practices-Kubernetes-Deployment.md [hunk 4]
```diff
@@ -205,7 +205,15 @@ Reference [here](http://kubernetes.io/docs/api-reference/v1/definitions/#_v1_pod
 
 
 
-In case you are running containers with elevated privileges (--privileged) you should consider using the “DenyEscalatingExec” admission control. This control denies exec and attach commands to pods that run with escalated privileges that allow host access. This includes pods that run as privileged, have access to the host IPC namespace, and have access to the host PID namespace. For more details on admission controls, see the Kubernetes [documentation](http://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/).
+By default, users with the ability to create pods can make use of features that allow root-level 
+access to the nodes they run on, including mounting arbitrary host paths, running as root, and
```
Comment (zparnold) [inline content/en/blog/_posts/2016-08-00-Security-Best-Practices-Kubernetes-Deployment.md]:
```suggestion
access to the Nodes they run on, including mounting arbitrary host paths, running as the `root` user, and
```

File: content/en/blog/_posts/2016-08-00-Security-Best-Practices-Kubernetes-Deployment.md [hunk 5]
```diff
@@ -205,7 +205,15 @@ Reference [here](http://kubernetes.io/docs/api-reference/v1/definitions/#_v1_pod
 
 
 
-In case you are running containers with elevated privileges (--privileged) you should consider using the “DenyEscalatingExec” admission control. This control denies exec and attach commands to pods that run with escalated privileges that allow host access. This includes pods that run as privileged, have access to the host IPC namespace, and have access to the host PID namespace. For more details on admission controls, see the Kubernetes [documentation](http://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/).
+By default, users with the ability to create pods can make use of features that allow root-level
```
Comment (zparnold) [inline content/en/blog/_posts/2016-08-00-Security-Best-Practices-Kubernetes-Deployment.md]:
```suggestion
By default, users with the ability to create Pods can make use of features that allow root-level 
```

File: content/en/docs/reference/access-authn-authz/admission-controllers.md [hunk 6]
```diff
@@ -142,20 +142,24 @@ if the pods don't already have toleration for taints
...
 This admission controller will intercept all requests to exec a command in a pod if that pod has a privileged container.
 
-If your cluster supports privileged containers, and you want to restrict the ability of end-users to exec
-commands in those containers, we strongly encourage enabling this admission controller.
-
 This functionality has been merged into [DenyEscalatingExec](#denyescalatingexec).
+The DenyExecOnPrivileged admission plugin is deprecated and will be removed in v1.18.
+
+Use of a policy-based admission plugin (like [PodSecurityPolicy](#podsecuritypolicy) or a custom admission plugin)
+which can be targeted at specific users or namespaces, and also protects against creation of overly privileged pods
```
Comment (zparnold) [inline content/en/docs/reference/access-authn-authz/admission-controllers.md]:
```suggestion
which can be targeted at specific users or Namespaces and also protects against creation of overly privileged Pods
```

File: content/en/docs/reference/access-authn-authz/admission-controllers.md [hunk 7]
```diff
@@ -142,20 +142,24 @@ if the pods don't already have toleration for taints
...
 allow host access.  This includes pods that run as privileged, have access to the host IPC namespace, and
 have access to the host PID namespace.
 
-If your cluster supports containers that run with escalated privileges, and you want to
-restrict the ability of end-users to exec commands in those containers, we strongly encourage
-enabling this admission controller.
+The DenyEscalatingExec admission plugin is deprecated and will be removed in v1.18.
+
+Use of a policy-based admission plugin (like [PodSecurityPolicy](#podsecuritypolicy) or a custom admission plugin)
+which can be targeted at specific users or namespaces, and also protects against creation of overly privileged pods
```
Comment (zparnold) [inline content/en/docs/reference/access-authn-authz/admission-controllers.md]:
```suggestion
which can be targeted at specific users or Namespaces and also protects against creation of overly privileged Pods
```

**Comment by zparnold** [comment]
@liggitt Small nits, also great to see you again! Let me know if you (dis)agree. I made them as suggestions, so you should just be able to accept them if they work.

**Comment by liggitt** [comment]
@zparnold made the suggested changed inline, and reverted the blog post content changes in favor of a note at the top pointing to latest cluster hardening guides.

**Comment by Rajakavitha1** [comment]
@zparnold Please take a look and state your approval.

**Pull request merged**